File: egghead.io/data-structures-and-algorithms-in-js/lesson-10-build-a-bt-in-js-and-traversal/node.py
class Node:
  def __init__(self, data):
    self.data = data
    self.prev = None
    self.succ = None

  def add_node(self, data):
    node = Node(data)

    if data < self.data:
      self.prev = node
    else:
      self.succ = node
    
    return node

File: egghead.io/data-structures-and-algorithms-in-js/lesson-10-build-a-bt-in-js-and-traversal/node.js
class Node {
  constructor(data) {
    this.data = data;
    this.prev = null;
    this.next = null;
  }

  addNode(data) {
    const node = new Node(data);

    if (data < this.data) {
      this.prev = node;
    } else {
      this.next = node;
    }

    return node;
  }
}

module.exports = Node;
